setup.cfg

```diff
@@ -3,7 +3,6 @@ addopts =
     --mypy
     --doctest-modules
     --ignore=esmvalcore/cmor/tables/
-    --cov=esmvalcore
```

Comment:
OK, I like the idea of buying us time via not cov-outputting at CI level, but how much time is it saved at user-level? I find it very useful to have the coverage report immediately available after all tests have run, and I bet you my booths if one is not a serious developer, always interested in the coverage report, then they don't even both running the tests. So I'd say remove it from the CI via `--no-cov` flag, but leave it for the user - what say you, bud?

Reply:
I usually only look at the coverage report provided by Codecov because it shows me the relevant coverage changes as opposed to everything. I also run the tests way more often to check if I did not mess things up, e.g. before committing, then to check if I covered every changed line of code with tests. But I admit this is just a personal preference. 

Let's ask the others: @ESMValGroup/technical-lead-development-team What is your experience with code coverage? Would you prefer to have it on or off on the command line by default? i.e. should the default be `pytest --cov` or `pytest --no-cov`? Running the tests with coverage takes 38 seconds on my laptop, while running them without coverage takes 24 seconds.

Reply:
we may want to use `--testmon` and have both jam and peanut butter, it looks like the previous issues with `--cov` and `--testmon` have now been solved https://github.com/tarpas/pytest-testmon/issues/86 NOTE: testmon allows coverage production/updating only for the bits of code that changed

Reply:
> Running the tests with coverage takes 38 seconds on my laptop, while running them without coverage takes 24 seconds.

Wow, it's more like ~3 minutes and ~2 minutes on my machine! 😮

My preference would be to have coverage run via a GitHub workflow, but not locally (since I like to test locally frequently, and not for coverage reasons, but I would like to make sure my coverage is good when creating a PR) 👍

Reply:
Thanks @bouweandela, `pytest -n auto` used 4 workers and reduced the timings to ~2 minutes and ~1 minute 👍 (I'm guessing my Linux machine isn't as good as yours!) 😝

Reply:
nobody make comments about Linux boxes, my Ubuntu 18.04 is flying through those tests :rofl:

Reply:
Fine, upon me growing older and lazier